src/lib/proof-system/zkprogram.ts

```diff
@@ -1230,16 +1353,33 @@ function computeFeatureFlags(gates: Gate[]): MlFeatureFlags {
     let flag = gateToFlag[gate.type];
     if (flag !== undefined) flags[flag] = true;
   }
+  return flags;
+}
+
+function featureFlagsToMlOption(
```

Comment:
I removed the old binary `true | false` feature flag data structure and replaced it entirely with the new option one, also in `picklesRuleFromFunction` so we dont have to maintain multiple of these conversion helpers